ArcGISRuntimeSDKQt_CppSamples/Maps/ApplyScheduledMapUpdates/ApplyScheduledMapUpdates.cpp

```diff
@@ -0,0 +1,216 @@
+  QObject(parent)
+{
+  // For the purposes of demonstrating the sample,
+  // create a temporary copy of the local offline map files,
+  // so that updating does not overwrite them permanently
+  copyDir(defaultDataPath() + "/" + sampleMmpk, m_TempDir.path() + "/" + sampleMmpk);
+}
+
+ApplyScheduledMapUpdates::~ApplyScheduledMapUpdates()
+{
+  if (m_mobileMapPackage)
+  {
+    m_mobileMapPackage->close();
+  }
+}
+
+void ApplyScheduledMapUpdates::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ApplyScheduledMapUpdates>("Esri.Samples", 1, 0, "ApplyScheduledMapUpdatesSample");
+}
+
+void ApplyScheduledMapUpdates::updateMap()
+{
+  if (!m_offlineSyncTask)
+    return;
+
+  m_offlineSyncTask->createDefaultOfflineMapSyncParameters();
+}
+
+MapQuickView* ApplyScheduledMapUpdates::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ApplyScheduledMapUpdates::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+
+  // create MMPK
+  qDebug() << m_TempDir.path() + "/" + sampleMmpk;
+  m_mobileMapPackage = new MobileMapPackage(m_TempDir.path() + "/" + sampleMmpk, this);
+
+  // load MMPK
+  connect(m_mobileMapPackage, &MobileMapPackage::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+      return;
+
+    // set the map on the map view
+    m_map = m_mobileMapPackage->maps().at(0);
+    m_mapView->setMap(m_map);
+
+    // setup sync task
+    m_offlineSyncTask = new OfflineMapSyncTask(m_map, this);
+
+    // connect sync task signals
+    connectSyncSignals();
+
+    // check for updates
+    m_offlineSyncTask->checkForUpdates();
+  });
+
+  m_mobileMapPackage->load();
+
+  emit mapViewChanged();
+}
+
+void ApplyScheduledMapUpdates::connectSyncSignals()
+{
+  // connect to checkForUpdatesCompleted signal and update the UI accordingly
+  connect(m_offlineSyncTask, &OfflineMapSyncTask::checkForUpdatesCompleted, [this](QUuid, OfflineMapUpdatesInfo* info)
+  {
+    if (info->downloadAvailability() == OfflineUpdateAvailability::Available)
+    {
+      emit updateUi(true, "Updates Available", QString("Updates size: %1 bytes").arg(info->scheduledUpdatesDownloadSize()));
+    }
+    else
+    {
+      emit updateUi(false, "No updates available", "The preplanned map area is up to date");
+    }
+
+    delete info;
+  });
+
+  // connect to createDefaultOfflineMapSyncParametersCompleted signal
+  connect(m_offlineSyncTask, &OfflineMapSyncTask::createDefaultOfflineMapSyncParametersCompleted, [this](QUuid, OfflineMapSyncParameters parameters)
+  {
+    // set the parameters to download all updates for the mobile map packages
+    parameters.setPreplannedScheduledUpdatesOption(PreplannedScheduledUpdatesOption::DownloadAllUpdates);
+
+    // set the map package to rollback to the old state should the sync job fail
+    parameters.setRollbackOnFailure(true);
```

Comment:
this setting is invalid for scheduled updates - we should remove this from the sample and make sure it's not in the common sample design I guess...

Reply:
Ok, I'll notify the team. It was in the Java sample which I based this off of